pkg/controller/account/account_controller.go

```diff
@@ -617,9 +629,29 @@ func (r *ReconcileAccount) initializeRegions(reqLogger logr.Logger, currentAcctI
 		return err
 	}
 
+	// Instantiate a client with a default region to retrieve regions we want to initialize
+	awsClient, err := r.awsClientBuilder.GetClient(controllerName, r.Client, awsclient.NewAwsClientInput{
+		AwsCredsSecretIDKey:     *creds.Credentials.AccessKeyId,
+		AwsCredsSecretAccessKey: *creds.Credentials.SecretAccessKey,
+		AwsToken:                *creds.Credentials.SessionToken,
+		AwsRegion:               awsv1alpha1.AwsUSEastOneRegion,
+	})
+	if err != nil {
+		connErr := fmt.Sprintf("unable to connect to default region %s", awsv1alpha1.AwsUSEastOneRegion)
+		reqLogger.Error(err, connErr)
+	}
+
+	regionsEnabledInAccount, err := awsClient.DescribeRegions(&ec2.DescribeRegionsInput{
+		AllRegions: aws.Bool(false),
+	})
+	if err != nil {
+		reqLogger.Error(err, "Failed to retrieve list of regions enabled in this account.")
+		return err
+	}
+
 	// This initializes supported regions, and updates Account state when that's done. There is
 	// no error checking at this level.
-	go r.asyncRegionInit(reqLogger, currentAcctInstance, creds)
+	go r.asyncRegionInit(reqLogger, currentAcctInstance, creds, regionAMIs, regionsEnabledInAccount)
```

Comment:
NOOP for today: I feel like it would be cleaner to just parse the list of regions enabled and only pass the list of AMIs that it needs to in here, but that's a large refactor here, and can be tackled later.  Let's just keep that in mind.

Reply:
I agree with you, the only reason I did it this way is that it is the last layer in which I can error out normally. Happy to change it in the future though.